packages/components/scrollbar/src/scrollbar.vue

```diff
@@ -140,19 +140,15 @@ export default defineComponent({
       sizeWidth.value = width + GAP < offsetWidth ? `${width}px` : ''
     }
 
-    watch(
-      () => props.noresize,
-      (noresize) => {
-        if (noresize) {
-          stopResizeObserver?.()
-          stopResizeListener?.()
-        } else {
-          ;({ stop: stopResizeObserver } = useResizeObserver(resize$, update))
-          stopResizeListener = useEventListener('resize', update)
-        }
-      },
-      { immediate: true }
-    )
+    watchEffect(() => {
+      if (!props.noresize && !props.native) {
+        ;({ stop: stopResizeObserver } = useResizeObserver(resize$, update))
+        stopResizeListener = useEventListener('resize', update)
+      } else {
+        stopResizeObserver?.()
+        stopResizeListener?.()
+      }
+    })
```

Comment:
**Praise:** the code seems legit to me, would you mind adding a test for your code? If you don't know how to do that, I can assist you with it.

Reply:
😀 ，I know little about testing. I would appreciate it if you could help me.